cunumeric/array.py

```diff
@@ -1754,6 +1754,10 @@ def argmax(
             raise ValueError("output array must have int64 dtype")
         if axis is not None and not isinstance(axis, int):
             raise ValueError("axis must be an integer")
+        if axis is not None and abs(axis) > self.ndim:
+            raise ValueError(
+                "axis must be smaller than array.ndim and bigger than 0 "
+            )
```

Comment:
Numpy seems to accept negative axis values:
```
In [2]: a = np.array([1,1,2,3, 1])

In [3]: a.argmax(axis=-1)
Out[3]: 3
```
Is no negative axis values a limitation of cunumeric? 

Also for out of bounds it raises `AxisError` not `ValueError`: 
```
In [4]: a.argmax(axis=19)
---------------------------------------------------------------------------
AxisError                                 Traceback (most recent call last)
Cell In [4], line 1
----> 1 a.argmax(axis=19)

AxisError: axis 19 is out of bounds for array of dimension 1
```
we sholud match.

Reply:
fix it in new commit.

Reply:
It looks like you deleted the check entirely? I don't think that is correct: `abs(axis) > self.ndim` *is* an error, but it is an `AxisError` not a `ValueError`